sdc-plugin/tests/get_clocks/get_clocks.golden.txt

```diff
@@ -0,0 +1,6 @@
+{$auto$clkbufmap.cc:247:execute$1917} {$auto$clkbufmap.cc:247:execute$1919} clk clk2 clk_int_1 middle_inst_1.clk_int middle_inst_4.clk
```

Comment:
There doesn't appear to be a difference between line 1 and 2?  What is the expected behavior of `-include_generated_clocks`, because I don't see you actually testing that?

Reply:
I think we may have a conceptual interface mismatch with how Vivado treats clock objects.  The clock object on both sides of a IBUF and BUFG are the same clock object.  So in this example, there should really just be 1 clock `clk`, which has the nets listed on line 1 as the nets that are on that clock.  PLL's / MMCM's and BUFGCTRL's do split clock definitions because the underlying properties of the clock change across the elements.  The propigation that occurs should probably be something akin too:
 - Does this element change the clock -> No:
   - If the output net has no explicit clock membership, mark the net as being part of the same clock
 - Does this element change the clock -> Yes:
  - If the output net has an explicit clock definition, do nothing
  - If the output net has no explicit clock definition, create a new generated clock with the new properties based on the elements configuration (e.g. PLL creates a new clock with period X, waveform Y Z).

Does that make sense?

Reply:
To phrase it a different way:
 - Clock objects has properties (e.g. waveform, period, is generated)
 - Clocks can have relationships to each other (e.g. false path, jitter, https://docs.verilogtorouting.org/en/latest/vpr/sdc_commands/#set-max-delay-set-min-delay)
 - Nets may have a clock object they are associated with.  The propigation step should associate nets with their clocks as needed (e.g. propagate through IBUF/BUFG).
 - Some cells (PLL/MMCM) define generated clocks using a relationship between the source clock and output clock.  The propagate through PLL/MMCM should generate these related clocks.

Reply:
> There doesn't appear to be a difference between line 1 and 2? What is the expected behavior of -include_generated_clocks, because I don't see you actually testing that?

Currently there is no difference between these two lines as all clocks are treated as non-generated, hence they are never skipped if the -include_generated_clocks is not specified.
Once issue #53 is implemented there will be differences, but for now we simply test the command interface.

Reply:
@litghost I agree with the clock concept you described above. I opened issue #53 to capture what you described as I believe it's beyond the scope of this PR.